lib/ansible/plugins/inventory/online.py

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: online
+    plugin_type: inventory
+    author:
+      - Remy Leone (@sieben)
+    short_description: Online inventory source
+    description:
+        - Get inventory hosts from Online
+    options:
+        plugin:
+            description: token that ensures this is a source file for the 'online' plugin.
+            required: True
+            choices: ['online']
+        oauth_token:
+            required: True
+            description: Online OAuth token.
+            env:
+                # in order of precedence
+                - name: ONLINE_TOKEN
+                - name: ONLINE_API_KEY
+                - name: ONLINE_OAUTH_TOKEN
+        hostnames:
+            description: List of preference about what to use as an hostname.
+            type: list
+            default:
+                - public_ipv4
+            choices:
+                - public_ipv4
+                - private_ipv4
+                - hostname
+'''
+
+EXAMPLES = '''
+# online_inventory.yml file in YAML format
+# Example command line: ansible-inventory --list -i online_inventory.yml
+
+plugin: online
+hostnames:
+  - public_ipv4
+'''
+
+import json
+
+from ansible.errors import AnsibleError
+from ansible.module_utils.urls import open_url
+from ansible.plugins.inventory import BaseInventoryPlugin
+from ansible.module_utils._text import to_native, to_text
+from ansible.module_utils.six.moves.urllib.parse import urljoin
+
+
+def _fetch_information(url, token):
+    headers = {
+        'Authorization': "Bearer %s" % token,
+        'Content-type': 'application/json'
+    }
+    try:
+        response = open_url(url, headers=headers)
+    except Exception as e:
+        raise AnsibleError("Error while fetching %s: %s" % (url, to_native(e)))
+
+    try:
+        raw_data = to_text(response.read(), errors='surrogate_or_strict')
```

Comment:
This seems to be related to https://github.com/ansible/ansible/pull/43715. Can we get a final agreement on the other PR first before pushing that here?